robot/impl/local_robot.go

```diff
@@ -791,7 +791,7 @@ func (r *localRobot) getOptionalDependencies(conf resource.Config) resource.Depe
 	optDeps := make(resource.Dependencies)
 
 	for _, optionalDepNameString := range conf.ImplicitOptionalDependsOn {
-		matchingResourceNames := r.manager.resources.FindNodesByShortName(optionalDepNameString)
+		matchingResourceNames := r.manager.resources.FindBySimpleName(optionalDepNameString)
```

Comment:
FindNodesByShortName will not match on prefixes, but FindBySimpleName does.

don't think we should be using FindNodesByShortName anymore, since robots shouldn't be using ShortNames anymore